proposals/0035-linalg-matrix.md

```diff
@@ -70,47 +89,758 @@ class Matrix {
     requires ArithmeticScalar<T>
   Matrix operator/(T);
 
-  static Matrix Splat(ElTy Val);
-  static Matrix Load(ByteAddressBuffer Res, uint StartOffset, uint Stride,
-                     bool ColMajor, uint Align = sizeof(ComponentTy));
-  static Matrix Load(RWByteAddressBuffer Res, uint StartOffset, uint Stride,
-                     bool ColMajor, uint Align = sizeof(ComponentTy));
+  // Apply a unary operation to each element.
+  template<UnaryOperation Op>
+  Matrix ApplyUnaryOperation();
+
+  template <typename T>
+    requires ArithmeticScalar<T>
+  static Matrix Splat(T Val);
+  static Matrix
+  Load(ByteAddressBuffer Res, uint StartOffset, uint Stride, bool ColMajor,
+       uint Align = sizeof(ElementType));
+  static Matrix
+  Load(RWByteAddressBuffer Res, uint StartOffset, uint Stride, bool ColMajor,
+       uint Align = sizeof(ElementType));
 
-  static Matrix Load(groupshared ElTy Arr[], uint StartIdx, uint Stride,
+  template <typename T>
+    requires ArithmeticScalar<T>
+  static Matrix Load(groupshared T Arr[], uint StartIdx, uint Stride,
                      bool ColMajor);
 
-  void Store(RWByteAddressBuffer Res, uint StartOffset, uint Stride,
-             bool ColMajor, uint Align = sizeof(ComponentTy));
+  void
+  Store(RWByteAddressBuffer Res, uint StartOffset, uint Stride, bool ColMajor,
+        uint Align = sizeof(ElementType));
+
+  template <typename T>
+    requires ArithmeticScalar<T>
+  void Store(groupshared T Arr[], uint StartIdx, uint Stride,
+             bool ColMajor);
+
+  // Row accesses
+  vector<ElementType, M> GetRow(uint Index);

```

Comment:
Should we have an (expensive?) `GetCol(uint Index)` function for a complete API

Reply:
I'll add this into the outstanding questions section.